Gems/AtomLyIntegration/CommonFeatures/Code/Include/AtomLyIntegration/CommonFeatures/Mesh/AtomMeshBus.h

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/EBus/EBus.h>
+#include <AzCore/Component/EntityId.h>
+#include <Atom/Feature/Mesh/MeshFeatureProcessorInterface.h>
+
+namespace AZ
+{
+    namespace Render
+    {
+        //! Bus for retrieving data about a given entity's Atom mesh.
+        class AtomMeshRequests
+        : public AZ::EBusTraits
+        {
+        public:
+            static const AZ::EBusHandlerPolicy HandlerPolicy = AZ::EBusHandlerPolicy::Multiple;
+            static const EBusAddressPolicy AddressPolicy = EBusAddressPolicy::ById;
+            using BusIdType = EntityId;
+
+            //! Returns the handle to the Atom mesh.
+            virtual const MeshFeatureProcessorInterface::MeshHandle* GetMeshHandle() const = 0;
+        };
+
+        using AtomMeshRequestBus = EBus<AtomMeshRequests>;
+        
+        //! Bus for receiving notifications about a given entity's Atom mesh state.
+        class AtomMeshNotifications
+        : public AZ::EBusTraits
+        {
+        public:
+            static const AZ::EBusHandlerPolicy HandlerPolicy = AZ::EBusHandlerPolicy::Multiple;
+            static const EBusAddressPolicy AddressPolicy = EBusAddressPolicy::ById;
+            using BusIdType = EntityId;
+
+            //! Notification for when the Atom mesh handle has been acquired (and thus ready for use).
+            virtual void OnAcquireMesh(const MeshFeatureProcessorInterface::MeshHandle* meshHandle) = 0;
```

Comment:
Consider renaming this OnMeshHandleSet because it should probably handle the case that the mesh handle is cleared.

Reply:
Great suggestion, will change 👍